tracing-test/src/span.rs

```diff
@@ -8,35 +8,33 @@ use std::fmt;
 /// `subscriber` module.
 #[derive(Clone, Debug, Default, Eq, PartialEq)]
 pub struct MockSpan {
-    pub(in crate::support) metadata: metadata::Expect,
+    pub(crate) metadata: metadata::Expect,
 }
 
 #[derive(Debug, Default, Eq, PartialEq)]
 pub struct NewSpan {
-    pub(in crate::support) span: MockSpan,
-    pub(in crate::support) fields: field::Expect,
-    pub(in crate::support) parent: Option<Parent>,
+    pub(crate) span: MockSpan,
+    pub(crate) fields: field::Expect,
+    pub(crate) parent: Option<Parent>,
 }
 
-pub fn mock() -> MockSpan {
+pub fn span<I>(name: I) -> MockSpan
+where
+    I: Into<String>,
+{
     MockSpan {
-        ..Default::default()
+        metadata: metadata::Expect {
+            name: Some(name.into()),
+            ..Default::default()
+        },
     }
 }
 
-impl MockSpan {
-    pub fn named<I>(self, name: I) -> Self
-    where
-        I: Into<String>,
-    {
-        Self {
-            metadata: metadata::Expect {
-                name: Some(name.into()),
-                ..self.metadata
-            },
-        }
-    }
+pub fn new() -> MockSpan {
```

Comment:
WDYT about naming this something like `any`? Then, the test DSL would look like this:
```rust
span::any()
   .at_level(...)
   .with_field(...)
```